### ISSUE TITLE
Flatpak release for Linux

### ISSUE THREAD

**Issue by 12people** [vassalengine/vassal#11853]
It'd be great if Vassal could offer Flatpak releases for Linux.

Flatpak is an increasingly popular sandboxed and distro-independent way of packaging apps for Linux. It's bundled with a large number of distros, from Fedora to Pop! OS to Linux Mint, and is the only way to install apps on immutable operating systems like Steam OS or Fedora Silverblue.

Here's the [documentation for building Flatpaks](https://docs.flatpak.org/en/latest/introduction.html). And here's an example of a [Java app's Flatpak](https://github.com/flathub/org.enjoyingfoss.FOSStriangulator/blob/master/org.enjoyingfoss.FOSStriangulator.yml).

As a Flatpak, Vassal could then be released on [Flathub](https://flathub.org/), the most popular repository of Flatpak apps.

**Comment by uckelman** [comment]
I'd be willing to review it were you to submit a PR.

**Comment by BrentEaston** [comment]
Hi @12people, Sounds like a great idea, but we have limited resources to do this at the moment. Do you have the tech skills to set this up?

**Comment by 12people** [comment]
Hi @BrentEaston,
I've been a bit short on time lately, but I can take a look at it when I find the time to.

**Comment by lmcx** [comment]
Hi,

I had the same need so I made a very simple flatpak manifest for VASSAL. It can be found here : https://github.com/lmcx/org.vassalengine.vassal

@uckelman I would recommend to create a new repo for that and not push it to the source one.

This simple flatpak manifest has been tested on Fedora Silverblue 37 with VASL 6.6.5.1. It installs VASSAL with Flatpak, creates a .desktop file and make it visible with its .svg icon. Modules loading and online games seem to work fine.

It can be tested with :
```
$ flatpak-builder --user --install --force-clean build org.vassalengine.vassal.yml
$ flatpak run org.vassalengine.vassal
```
The second command is not mandatory since Vassal should be available in the user's software list after the first one.

You can then uninstall it with :
`$ flatpak uninstall org.vassalengine.vassal`

Feel free to take the code and use it at your convenience. Though, I'd be happy to help to get it to Flathub if needed.

**Comment by lmcx** [comment]
Quick update : I modified the manifest so it builds from source instead of downloading the binary.
This would allow to implement [portals](https://github.com/flatpak/flatpak/wiki/Portals) through a patch, instead of giving direct access to the user's home folder.

I am not a good enough java dev to implement that by myself though, if that seems like something Vassal's devs would like to see I could help with the "Flatpak part" but might need to be guided through the code.

**Comment by 12people** [comment]
Thanks for the Flatpak! I created a similar one, just didn't have time to test it.

As for portals, it should be possible to implement those via Gnome's ["Zenity" shell program](https://github.com/GNOME/zenity) — see this [Reddit thread for info](https://www.reddit.com/r/gnome/comments/lws7w9/does_javafx_filechooser_make_use_of_portals/).

**Comment by lmcx** [comment]
Portals are highly recommended in Flathub publishing [requirements](https://github.com/flathub/flathub/wiki/App-Requirements#filesystem-access) so I don't know if it is such a good idea to separate the two subjects.

ATLauncher's [FileChooserDialog](https://github.com/ATLauncher/ATLauncher/blob/master/src/main/java/com/atlauncher/gui/dialogs/FileChooserDialog.java) could be a better source of inspiration for portals.

I guess most of the work could be done in [FileChooser.java](https://github.com/vassalengine/vassal/blob/master/vassal-app/src/main/java/VASSAL/tools/filechooser/FileChooser.java) with a DBusFileChooser class extending FileChooser and instantiated in createFileChooser() if "/.flatpak-info" exists on the system.

This should probably not be done in the main source code but, instead, through a patch in the Flatpak manifest.

I have been trying to implement this class, but due to my lack of experience with Java it has been a quite painful task so far. This might require an actual Java dev to be implemented properly.

Even with this class done, there is still a solution to find for the $HOME/.VASSAL/ directory and its content, which are not accessed via FileChooser.

**Comment by 12people** [comment]
@lmcx 

As long as there is a contributor willing to implement it, I'm all for it.
If not, though, I'm worried the Flatpak will never be released.

If there isn't anyone here actively working on it, I still think it would be better to do this piecemeal and release the Flatpak first, then make file portal changes.

**Comment by lmcx** [comment]
@12people I just upgraded to 3.6.17, would you have time to check if you see anything weird in the manifest and patch and try and play with the flatpak a little before I submit a PR to Flathub ?

**Comment by 12people** [comment]
Unfortunately, I've been finding it hard to find free time lately. :(

So I unfortunately don't have time to test this.

However, I did quickly look at the files in your repo:

**[The Desktop file](https://github.com/lmcx/org.vassalengine.vassal/blob/main/org.vassalengine.vassal.desktop)**
- The "Version" doesn't pertain to the app, but rather to the version of the desktop file metadata. Therefore, it should be "Version=1.0"
- I would remove the "MimeType=" line.

**[The appdata file](https://github.com/lmcx/org.vassalengine.vassal/blob/main/org.vassalengine.vassal.metainfo.xml)**
- I would recommend adding at least these two lines:
    - `<url type="homepage">https://vassalengine.org/</url>`
    - `<url type="bugtracker">https://github.com/vassalengine/vassal/issues/</url>`
- I think Flathub might require `releases` and `content_rating` to also be provided:
    - `<content_rating type="oars-1.1" />` (I'm not 100% sure it qualifies for OARS 1.1, though. Might be good to check.)
    - `<releases><release version="3.6.17" date="2023-04-24" /></releases>`

**[The Flatpak manifest](https://github.com/lmcx/org.vassalengine.vassal/blob/main/org.vassalengine.vassal.yml)**
- I don't think Flathub maintainers will allow you to have extra files in the repo reserved for the manifest. I think you'll need to create a PR for those files here so that they're bundled along with the source code. That would mean you'd only have the git repo in sources and would need to change your build commands to reflect that.
- Flathub definitely doesn't allow access to the network during the build process. So you'll need to remove `build-args: - --share=network`. Do you need it, though?

**Comment by lmcx** [comment]
@12people 
> Unfortunately, I've been finding it hard to find free time lately. :(
> 
> So I unfortunately don't have time to test this.

That might be better to wait for someone else to test it before submitting a PR. I am not really comfortable with the idea of releasing without properly testing.

> However, I did quickly look at the files in your repo:
> [...]

Great suggestions, thanks. I fixed all that.

> **[The Flatpak manifest](https://github.com/lmcx/org.vassalengine.vassal/blob/main/org.vassalengine.vassal.yml)**
> 
>     * I don't think Flathub maintainers will allow you to have extra files in the repo reserved for the manifest. I think you'll need to create a PR for those files here so that they're bundled along with the source code. That would mean you'd only have the git repo in sources and would need to change your build commands to reflect that.

I am not sure to completely understand what you are talking about. If you have the `org.vassalengine.vassal.desktop` file in mind, it seems to be quite common to provide it that way (e.g.: https://github.com/flathub/com.google.ChromeDev).

If you are talking about the `vassal` shell script, I guess it could be done with a patch on https://github.com/vassalengine/vassal/blob/master/dist/VASSAL.sh instead, and then would not need to push the file anywhere.

>     * Flathub definitely doesn't allow access to the network during the build process. So you'll need to remove `build-args: - --share=network`. Do you need it, though?

Sadly, when disabling it Maven gets really grumpy :

```
patching file Makefile
Running: make release-linux
./mvnw versions:set -DnewVersion=3.6.17 -DgenerateBackupPoms=false
Exception in thread "main" java.net.UnknownHostException: repo.maven.apache.org
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:567)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
	at java.base/java.net.Socket.connect(Socket.java:633)
	at java.base/sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:304)
	at java.base/sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:174)
	at java.base/sun.net.NetworkClient.doConnect(NetworkClient.java:183)
	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:531)
	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:636)
	at java.base/sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:266)
	at java.base/sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:380)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:193)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1242)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1128)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:179)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1665)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1589)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:224)
	at org.apache.maven.wrapper.DefaultDownloader.downloadInternal(DefaultDownloader.java:95)
	at org.apache.maven.wrapper.DefaultDownloader.download(DefaultDownloader.java:85)
	at org.apache.maven.wrapper.Installer.createDist(Installer.java:84)
	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:160)
	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:73)
make: *** [Makefile:101: version-set] Error 1
Error: module vassal: Child process exited with code 2
```
I checked and you are right: network is disabled by Flathub during build. So there might be another way to bring these dependencies to the build process, but as previously said I am not a Java dev (and, to be completely honest, don't plan on becoming one) and am clearly juggling in the dark with that part.

As long as nobody with strong enough Java/Maven/... skills is OK to provide some help, I won't invest more time in that, sorry.

**Comment by uckelman** [comment]
One of Maven's jobs is to fetch dependencies. You're going to have a hard time doing that without internet access. Are there any Flatpack releases of other projects which use a dependency manager? A lot of Python projects use pip or poetry, and most Rust projects use cargo. Those also fetch dependencies. Could you check how they handle it?

**Comment by 12people** [comment]
@uckelman The way this is done is to download the dependencies straight from source, not via a dependency manager.

@lcmx I guess I don't see the reason why the app should be compiled via Flatpak. Unlike native apps, Java apps run in the Java Virtual Machine and therefore don't need to be compiled for a specific architecture. Can't you just ship the precompiled release package like you did in your initial Flatpak proposal?

As for the other files in your repository, I have first-hand experience of my first proposed app to Flathub being rejected for exactly this reason. It's true that there are indeed some apps on Flathub that have a few other files than the manifest in their repo. However, I've been told that this is done only for those apps where the upstream doesn't want to collaborate on an official Flatpak and thus won't include Flatpak-specific files in their repo. (Such as Google Chrome.)

**Comment by lmcx** [comment]
@uckelman @12people There is a PoC on my repo, using the maven-dependencies.yml file to pre-fetch everything before the build.

@12people Those three subjects are actually linked to a single question : is vassalengine team planning to incorporate Flatpak's specific stuff in their code ?
I'm not trying to rebuild VASSAL based on a misunderstanding about Java, but on the hypothesis that the answer to this question might be "no".

We obviously could ship the precompiled package, but I don't think we should without integrating portals so ... same question.

But if the answer is yes, and someone is ready to
- Write code for the FileChooser and Document portals (referenced in a previous reply to this issue, with an implementation example from ATlauncher);
- Adapt the launcher so it works inside flatpak's sandbox;
- Add a desktop entry.
Then yes, we should definitely ship the official release and not rebuild everything, and everything that's not the manifest and flathub metadata could then be removed.

Until then, my repo remains open for whomever would like to try and implement those things.

**Comment by NN708** [comment]
Hi everyone,

I noticed this issue and would like to see VASSAL available on Flathub. I have some experience maintaining other Flathub apps, so I'd like to share a few thoughts:

1. Flathub requires apps to be built from source (see [requirements](https://docs.flathub.org/docs/for-app-authors/requirements#building-from-source)). In this regard, @lmcx's manifest follows the correct approach.
2. Flathub also requires metadata files to be in the upstream repository (see [requirements](https://docs.flathub.org/docs/for-app-authors/requirements#required-metadata)). I see the desktop file was already added for Debian packaging in #14244, so we would primarily just need to add the AppStream MetaInfo file.
3. Regarding the file chooser portal, I agree with @12people that we can continue using home directory access for now, as implementing the portal would require non-trivial work.
4. As for the launcher script, it is specific to Flatpak, so I believe it's acceptable to keep it in the manifest repository.

Is anyone still interested in submitting a PR to Flathub? If not, I'd be happy to take care of it.

**Comment by NN708** [comment]
I have completed the PRs based on @lmcx's work. We are now waiting for https://github.com/vassalengine/vassal/pull/14352 to be merged before we can test building it in https://github.com/flathub/flathub/pull/7172.

**Comment by NN708** [comment]
A test build is available here: https://github.com/flathub/flathub/pull/7172#issuecomment-3569089425

**Comment by NN708** [comment]
Vassal is finally published on Flathub! Link: https://flathub.org/en/apps/org.vassalengine.vassal

Since #14352 hasn't been included in a stable release yet, the current Flatpak is built on 3.7.19-SNAPSHOT. We can switch to a stable version once it's available.

@uckelman, would you be interested in maintaining this officially? I can ask the Flathub team to grant you access to [flathub/org.vassalengine.vassal](https://github.com/flathub/org.vassalengine.vassal) repository. The maintenance work is minimal, primarily involving merging a PR to update the version and dependencies after a new release.

**Comment by NN708** [comment]
> I seem not to have commit rights for the [flathub/org.vassalengine.vassal](https://github.com/flathub/org.vassalengine.vassal) repo.

@uckelman An invitation has been sent to you (https://github.com/flathub/flathub/issues/7442). Please accept it within a week.

**Comment by uckelman** [comment]
It looks like 3.7.20 still isn't on flathub. I'm not sure what else I need to do.

**Comment by NN708** [comment]
@uckelman I think the Flatpak is already updated, but the metadata wasn't updated. The commit cf4e141f9319a26c96aa8c6298ddbef04e196743 needs to be applied before tagging the 3.7.20 release, or the tag will still contain the old metadata.

**Comment by uckelman** [comment]
I was never able to get the 3.7.20 build finished. We're on 3.7.21 now, but the build fails:

https://github.com/flathub-infra/vorarbeiter/actions/runs/24636596100/job/72033056283

**Comment by NN708** [comment]
> I was never able to get the 3.7.20 build finished. We're on 3.7.21 now, but the build fails:
> 
> https://github.com/flathub-infra/vorarbeiter/actions/runs/24636596100/job/72033056283

The 3.7.20 build succeeded and is already published (https://github.com/flathub/org.vassalengine.vassal/pull/6).

The 3.7.21 build is failing because you introduced `/tmp` access, which violates [this linter rule](https://docs.flathub.org/docs/for-app-authors/linter#finish-args-host-tmp-access).

**Comment by uckelman** [comment]
Are you sure about 3.7.20 having been published? I don't see it here: https://flathub.org/en/apps/org.vassalengine.vassal

It appears that this commit is using `/tmp`: https://github.com/vassalengine/vassal/commit/9090162ba25c719d4122f0651187a9366a9984d2

@cholmcc Do you have a suggestion about what to do here?

**Comment by NN708** [comment]
> Are you sure about 3.7.20 having been published? I don't see it here: https://flathub.org/en/apps/org.vassalengine.vassal

Yes, I've already explained this in the comment just above the previous one.

**Comment by uckelman** [comment]
I don't understand the explanation about 3.7.20, in that case. Why don't I see 3.7.20 [here](https://flathub.org/en/apps/org.vassalengine.vassal) if it's been published?

**Comment by NN708** [comment]
> I don't understand the explanation about 3.7.20, in that case. Why don't I see 3.7.20 [here](https://flathub.org/en/apps/org.vassalengine.vassal) if it's been published?

That's because the webpage you're seeing is generated from the source code tagged as `3.7.20`. However, if you check that commit, the release section still shows 3.7.19: https://github.com/vassalengine/vassal/blob/3.7.20/dist/linux/org.vassalengine.vassal.metainfo.xml

And since the metadata hasn't been properly updated in `3.7.21` either, the webpage will show 3.7.20 even after 3.7.21 is published.

**Comment by cholmcc** [comment]
> It appears that this commit is using `/tmp`: [9090162](https://github.com/vassalengine/vassal/commit/9090162ba25c719d4122f0651187a9366a9984d2)

The commit does _not_ use `/tmp` but the sub-directory `tmp` of the build directory.   That's actually not new to the commit - it was always used by the `make` of Vassal.  The difference is that the string `tmp` appears in the workflow. 

That said, the `org.vassalengine.vassal.yml` should have 

```
finish-args:
  ...
  - --filesystem=/tmp
```
because Vassal acccesses `/tmp` at run-time. 

As for the update of [org.vassalengine.vassal.metainfo.xml](https://github.com/vassalengine/vassal/blob/3.7.20/dist/linux/org.vassalengine.vassal.metainfo.xml):   The file should be updated before tagging the Vassal source tree for the release.   It seems like commits needs to be pushed before the tag.   In `developerguide.adoc` terms it should be 

❏ Commit the changes `git commit . -m "The x.y.z release"` .
❏ Push the changes `git push`.
❏ Tag current state as the release: `git tag -s -m 'The x.y.z release.' x.y.z`.
❏ Push tag to origin: `git push --tags` (this generates the release on GitHub). 

Otherwise, it seems like the commits done beforehand aren't propagated to the tag (see [here](https://therightstuff.medium.com/the-rights-and-wrongs-of-git-push-with-tags-998667eaed8f) and [here](https://stackoverflow.com/questions/3745135/push-git-commits-tags-simultaneously) for some thoughts on this).

BTW, [flathub/org.vassalengine.vassal/org.vassalengine.vassal.yml](https://github.com/flathub/org.vassalengine.vassal/blob/master/org.vassalengine.vassal.yml) is terrible out of date.  Take the recipe  generated as part of the release from the **flatpak-recipe** artefact ([here](https://github.com/vassalengine/vassal/actions/runs/24393805519)) - which also fixes the `/tmp` issue.   Delete the `patches` directory from the repo as they are no longer used. 

Yours,
_Christian_

**Comment by cholmcc** [comment]
> The 3.7.21 build is failing because you introduced `/tmp` access, which violates [this linter rule](https://docs.flathub.org/docs/for-app-authors/linter#finish-args-host-tmp-access).

Could you explain that a bit more?  The page linked to does not give a whole lot of information.  

The point is, Vassal _needs_ access to `/tmp` (and always has) so what is needed to in the YAML file for that to happen? 

Thanks.

Yours,
_Christian_

**Comment by uckelman** [comment]
> The point is, Vassal needs access to /tmp (and always has)

Really? Where?

**Comment by cholmcc** [comment]
> > The point is, Vassal needs access to /tmp (and always has)
> 
> Really? Where?

When unpacking images from help files, etc.  In fact, on start-up, Vassal creates a temporary directory in `/tmp` - see [here](https://github.com/vassalengine/vassal/blob/master/vassal-app/src/main/java/VASSAL/launch/StandardConfig.java#L88). 

Also, Vassal explicitly needs access to `~/` - or more specifically to `~/.VASSAL`.    It seems like two [exceptions](https://docs.flathub.org/docs/for-app-authors/linter#exceptions)  _must_ be enabled in the `flathub` set of the linter. 
```
{
    "errors": [
        "finish-args-home-filesystem-access",
        "finish-args-host-tmp-access"
    ],
    "message": "See https://docs.flathub.org/linter for details and exceptions"
}

```

Yours,
_Christian_

**Comment by cholmcc** [comment]
Thinking a bit more about it, it may be that one can avoid the disallowed permissions. Access to th real `/tmp` directory isn't needed and `~/.VASSAL` could be handled through `--persist` while documents could be accessed through `--xdg-documents`.

**Comment by cholmcc** [comment]
Replacing the permissions 
```
  - --filesystem=/tmp
  - --filesystem=home
```
with 
```
  - --persist=.VASSAL
  - --filesystem=xdg-documents
```
passes the linter check.  However, this has a number of consequences: 
- When running the app, the directory `~/.var/app/org.vassalengine.vassal/.VASSAL` will be mounted as `~/.VASSAL`.  That means that all VASSAL configurations and the `errorLog` isn't available in `~/.VASSAL` on the regular system, but as `~/.var/app/org.vassalengine.vassal/.VASSAL`. 
- Vassal can _only_ access files in `~/Documents` (or whatever `${XDG_DOCUMENTS_DIR}` points to), thus forcing users to store their modules, saves, extensions, logs, etc. in that directory or some sub-directory thereof. 
- That also means that users cannot set `VASSAL.conf` to anything they like - the directory specified _must_ either be under `~/.var/app/org.vassalengine.vassal/.VASSAL` or `~/Documents`. 

Obviously, if this is enabled, then that would need to be documented in various places. 

I can make a PR with this if you want. 

Yours,
_Christian_

**Comment by cholmcc** [comment]
> ❏ Commit the changes `git commit . -m "The x.y.z release"` . 
> ❏ Push the changes `git push`. 
> ❏ Tag current state as the release: `git tag -s -m 'The x.y.z release.' x.y.z`. 
> ❏ Push tag to origin: `git push --tags` (this generates the release on GitHub).

Turns out one doesn't really need the additional `push` - it should be 

❏ Commit the changes `git commit . -m "The x.y.z release"` . 
❏ Tag current state as the release: `git tag -s -m 'The x.y.z release.' x.y.z`. 
❏ Push commits and tag to origin: `git push --atomic origin HEAD x.y.z` (this generates the release on GitHub).

That has the benefit that the commits are pushed to the branch (`release-x.y`) together with the tag, and only _one_ workflow is triggered (`release.yml`).   Also, in the `developerguide.adoc`, instead of 

❏ `make MAVEN_VERSION=x.y.z` (This will update various `pom.xml` with the new version number.)

it should be 

❏ `make version-set MAVEN_VERSION=x.y.z` (This will update various `pom.xml` with the new version number.)

as there's no real need to compile the whole thing.   In fact, this step isn't really needed, because the workflow will set the version number explicitly.    However, it can be done just to keep the source up to date.    

Whether it should actually happen just before the release, or whether it is just as good to do it for the next release just _after_ release, is debatable.   It seems to me that there are lots of commits and pushed going on around a release (not least to sync `master` with the release branch), which could probably be avoided.   In fact, I'm not really sure I understand why there's a release branch in the first place - `release-3.7` and `master` seem to be in sync most of the time.  If one needs a bug-fix branch from a given tag - say `3.7.21`, one can always do `git checkout -b 3.7.21-fixes 3.7.21`. Maybe there's something I've missed. 

Please advice as to what you want for the FlatPak fixes: 
1. Do you want to use the more restrictive privileges - with associated user impact, but immediate deployment, or
2. Do you want to push for explicit linter exceptions, with a bit more lead time. 

Yours,
_Christian_

**Comment by NN708** [comment]
@cholmcc As per [Flathub requirements](https://docs.flathub.org/docs/for-app-authors/requirements#permissions), applications are required to use the minimum permissions.

Regarding your questions, please check the [current manifest](https://github.com/flathub/org.vassalengine.vassal/blob/master/org.vassalengine.vassal.yml) first. Note that:

1. `--filesystem=home` is already added as an exception (because #14366 is not implemented yet).
2. `--filesystem=/tmp` is not required for the app to work.

**Comment by cholmcc** [comment]
> [@cholmcc](https://github.com/cholmcc) As per [Flathub requirements](https://docs.flathub.org/docs/for-app-authors/requirements#permissions), applications are required to use the minimum permissions.

I know.   The question was what the FlatHub documentation really means - it is _very_ terse and not very informative. 

> 1. `--filesystem=home` is already added as an exception (because [Implement FileChooser portal #14366](https://github.com/vassalengine/vassal/issues/14366) is not implemented yet).

OK, great that `home` is allowed for `org.vassalengine.vassal` - removes a bunch of head-aches.   Incidentally, this whole notion of portals etc. is a pain in the behind and largely not need and even counter-productive.  It is as if FlatPak hasn't understood the privilege system of Un*x or BSD - something they have in common with Apple - and not in a good way. Yet another reason why I dislike FlatPak :smile: 

> 2\. `--filesystem=/tmp` is not required for the app to work.

Well, Vassal needs _a_ `/tmp` directory, but it does not need to be the _real_ `/tmp`.


> [current manifest](https://github.com/flathub/org.vassalengine.vassal/blob/master/org.vassalengine.vassal.yml)

The current manifest is _way_ out of date.    

- Take artefact **flatpak-recipe** from [here](https://github.com/vassalengine/vassal/actions/runs/24393805519)
- Unpack it into a clone of [flathub/org.vassalengine.vassal](https://github.com/flathub/org.vassalengine.vassal)
- Remove the `--filesystem=/tmp` permission from `org.vassalengine.vassal.yml` 
- Remove the `patches` directory as those are no longer needed 
- Commit 
- Push (and possibly make PR)

In the future,  `--filesystem=/tmp` will _not_ be in the recipe produced by the Vassal release workflow.   Also, the `org.vassalengine.org.metainfo.xml` should be in sync before [`vassalengine/vassal`](https://github.com/vassalengine/vassal) is tagged (@uckelman had a little funny procedure around that). 

Yours,
_Christian_

**Comment by NN708** [comment]
> * Take artefact **flatpak-recipe** from [here](https://github.com/vassalengine/vassal/actions/runs/24393805519)
> * Unpack it into a clone of [flathub/org.vassalengine.vassal](https://github.com/flathub/org.vassalengine.vassal)
> * Remove the `--filesystem=/tmp` permission from `org.vassalengine.vassal.yml`
> * Remove the `patches` directory as those are no longer needed
> * Commit
> * Push (and possibly make PR)

@cholmcc You can fork [flathub/org.vassalengine.vassal](https://github.com/flathub/org.vassalengine.vassal) and submit a PR with your changes directly.

**Comment by cholmcc** [comment]
See https://github.com/flathub/org.vassalengine.vassal/pull/11

**Comment by uckelman** [comment]
I've merged flathub/org.vassalengine.vassal/pull/11.

**Comment by uckelman** [comment]
The Flatpak build is failing again: https://github.com/flathub-infra/vorarbeiter/actions/runs/25282466613/job/74121803678

**Comment by cholmcc** [comment]
Seems like the flathub build environment is trying to pull in openjdk 26 with release 25.0.8, which doesn't make sense.  Perhaps the runtime versio should be left undefined and let flathub choose it itself.  Will give that a go.

**Comment by uckelman** [comment]
I thought we finally got a good build for 3.7.23, but 3.7.20 still appears on [flathub](https://flathub.org/en/apps/org.vassalengine.vassal).

**Comment by NN708** [comment]
> I thought we finally got a good build for 3.7.23, but 3.7.20 still appears on [flathub](https://flathub.org/en/apps/org.vassalengine.vassal).

This is not updated in real time. The delay can be up to one hour.

**Comment by uckelman** [comment]
3.7.23 is there now. Thanks!
